### PR TITLE
UIREQ-943 create Jest/RTL test for NoteCreateRoute.js

### PR DESCRIPTION
**Pull request by cchitneedi**
Refs: 
UIREQ-943 create Jest/RTL test for NoteCreateRoute.js

URL: https://issues.folio.org/browse/UIREQ-943

### CODE REVIEW

File: src/routes/NoteCreateRoute.test.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+const renderNoteCreateRoute = (data1, data2) => render(
+  <Router history={history}>
+    <NoteCreateRoute location={data1} history={data2} />
+  </Router>
+);
+describe('NoteCreateRoute', () => {
+  it('NoteCreatePage should render when location.state is not empty', () => {
+    renderNoteCreateRoute(locationData, historyData);
+    userEvent.click(screen.getByRole('button', { name: 'renderReferredRecord' }));
+    expect(screen.getByText('NoteCreatePage'));
```
Comment (artem-blazhko) [inline src/routes/NoteCreateRoute.test.js:60]: What do you expect in this test?

File: src/routes/NoteCreateRoute.test.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  'replace': () => {},
+  'go': () => {},
+  'goBack': () => {},
+  'goForward': () => {},
+  'block': () => {},
+  'listen': () => {},
+};
+
+const locationData = historyData.location;
+const renderNoteCreateRoute = (data1, data2) => render(
```
Comment (artem-blazhko) [inline src/routes/NoteCreateRoute.test.js:51]: Could you use a bit more specific names and avoid using "data1", "data2"?

File: src/routes/NoteCreateRoute.test.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    }
+  },
+  'createHref': () => {},
+  'push': () => {},
+  'replace': () => {},
+  'go': () => {},
+  'goBack': () => {},
+  'goForward': () => {},
+  'block': () => {},
+  'listen': () => {},
```
Comment (artem-blazhko) [inline src/routes/NoteCreateRoute.test.js:47]: I do not see any reason to wrap object properties in quotation marks.
Could you remove it?

File: src/routes/NoteCreateRoute.test.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import '../../test/jest/__mock__/stripesConfig.mock';
+import { Router } from 'react-router';
+import { createMemoryHistory } from 'history';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import NoteCreateRoute from './NoteCreateRoute';
+
+jest.mock('@folio/stripes/smart-components', () => ({
+  ...jest.requireActual('@folio/stripes/smart-components'),
+  NoteCreatePage: (props) => <div><span>NoteCreatePage</span><button type="button" onClick={() => props.renderReferredRecord()}>renderReferredRecord</button></div>,
```
Comment (artem-blazhko) [inline src/routes/NoteCreateRoute.test.js:12]: Could you add mock to 'ui-requests\test\jest\__mock__\stripesSmartComponents.mock.js'?

**Review by artem-blazhko — CHANGES_REQUESTED** [review]
Could add appropriate changes to CHANGELOG.md file?
Could you test redirect functionality in this file?

**Comment by cchitneedi** [comment]
Hi @artem-blazhko ,
Changes have been done as mentioned.

Thanks and Regards,
Charan

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,10 @@
 # Change history for ui-requests
 
 ## IN PROGRESS
+## [8.0.2](https://github.com/folio-org/ui-requests/tree/v8.0.2) (2023-05-16)
+* Create Jest/RTL test for NoteCreateRoute.js Refs: UIREQ-943
+* Added new file in fixtures folder historyData.js file for static data
+* Added NoteCreatePage function to 'ui-requests\test\jest_mock_\stripesSmartComponents.mock.js'
```
Comment (artem-blazhko) [inline CHANGELOG.md:7]: Could you add only:
`Create Jest/RTL test for NoteCreateRoute.js. Refs: UIREQ-943.`
at the end of the "IN PROGRESS" section?

**Review by artem-blazhko — CHANGES_REQUESTED** [review]

File: src/routes/NoteCreateRoute.test.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+const locationData = historyDataProp.location;
+const renderNoteCreateRoute = (locationProps, historyProps) => render(
+  <Router history={history}>
+    <NoteCreateRoute location={locationProps} history={historyProps} />
+  </Router>
+);
+describe('NoteCreateRoute', () => {
+  it('NoteCreatePage should render when location.state is not empty', () => {
+    renderNoteCreateRoute(locationData, historyData);
+    userEvent.click(screen.getByRole('button', { name: 'renderReferredRecord' }));
```
[the remark this answers is not in the document]
Comment (cchitneedi) [reply src/routes/NoteCreateRoute.test.js]: Changes Done

**Review by artem-blazhko — APPROVED** [review]

**Pull request merged**